### PR TITLE
fix(dark-mode): fix bugs and improvements to UX

### PR DESCRIPTION
**Pull request by knajjars**
https://github.com/user-attachments/assets/cf4f1339-8815-416e-a872-d05babff6f9e

- Scroll overflow for create/update assistant
- Do not show error message when name is not typed for assistant
- Do not override files on file picker select
- Auto scroll to assistant step on "Next" or "Back"

**AI Description**

This pull request makes changes to the assistant settings and UI components. 

## Summary
The changes include:
- Adding the ability to disable assistant knowledge for specific agents.
- Updating the agent settings form to include data sources by default when creating a new agent.
- Enhancing the collapsible section component to smoothly scroll to the expanded content.
- Refactoring the assistant settings form to include a new "Update" button and rearranging the "Delete Assistant" button.

## Details
- **AgentRightPanel.tsx**: 
  - Introduces the `useAgentsStore` hook to manage the `disabledAssistantKnowledge` state for specific agents.
  - Updates the `useMemo` logic to retrieve tool metadata for Google Drive, Read Document, and Search File tools.
  - Adjusts the rendering of agent knowledge files and folders based on the new tool metadata structure.
- **AgentSettingsForm.tsx**: 
  - Adds the `useSearchParams` hook to determine the default step based on the presence of data sources in the URL parameters.
  - Updates the initial state of the `currentStep` to either "dataSources" or "define" based on the default step.
- **CreateAgent.tsx** and **UpdateAgent.tsx**: 
  - Wraps the `AgentSettingsForm` component in a container with a `className` of "overflow-y-auto" to enable vertical scrolling.
- **CollapsibleSection.tsx**: 
  - Imports the `useEffect` and `useRef` hooks from React.
  - Initializes a `ref` variable using `useRef` to reference the collapsible section element.
  - In the `useEffect` hook, when the `isExpanded` state is true, the `ref.current` element is scrolled into view smoothly using the `scrollIntoView` method.
- **chat.ts** and **stores/index.ts**: 
  - Updates the imports to include the `useAgentsStore` hook, which provides access to the `disabledAssistantKnowledge` state.
- **agentsSlice.ts**: 
  - Introduces a new `disabledAssistantKnowledge` array in the initial state.
  - Adds a new `setUseAssistantKnowledge` action to enable or disable assistant knowledge for specific agents.
  - Implements the `setUseAssistantKnowledge` action to update the `disabledAssistantKnowledge` array based on the provided `useKnowledge` flag and `agentId`.

### CODE REVIEW

File: src/interfaces/assistants_web/src/components/Agents/AgentSettings/AgentSettingsForm.tsx [hunk 1]
```diff
@@ -142,7 +145,11 @@ export const AgentSettingsForm: React.FC<Props> = ({
         isExpanded={currentStep === 'define'}
         setIsExpanded={(expanded) => setCurrentStep(expanded ? 'define' : undefined)}
       >
-        <DefineAssistantStep fields={fields} setFields={setFields} nameError={nameError} />
+        <DefineAssistantStep
+          fields={fields}
+          setFields={setFields}
+          nameError={!!fields.name ? nameError : undefined}
```
Comment (abimacarmes) [inline src/interfaces/assistants_web/src/components/Agents/AgentSettings/AgentSettingsForm.tsx:151]: `fields.name` is required - if we don't want to show the 'Assistant name is required' error message, we can just remove from `nameError`
```suggestion
          nameError={nameError}
```

File: src/interfaces/assistants_web/src/components/Agents/AgentRightPanel.tsx [hunk 2]
```diff
@@ -21,7 +22,10 @@ const AgentRightPanel: React.FC<Props> = () => {
   const [isDeletingFile, setIsDeletingFile] = useState(false);
   const [selectedIndex, setSelectedIndex] = useState(0);
   // TODO: (khalil) configure this to use Google drive files
```
Comment (abimacarmes) [inline src/interfaces/assistants_web/src/components/Agents/AgentRightPanel.tsx:24]: i think you can remove the TODO now?

File: src/interfaces/assistants_web/src/components/Agents/AgentRightPanel.tsx [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
 
 import { Transition } from '@headlessui/react';
+import { uniqBy } from 'lodash';
 import { useMemo, useState } from 'react';
 
 import { CitationsTab } from '@/components/Agents/CitationsTab';
 import { IconButton } from '@/components/IconButton';
 import { Banner, Button, Icon, Switch, Tabs, Text, Tooltip } from '@/components/Shared';
-import { TOOL_GOOGLE_DRIVE_ID } from '@/constants';
+import { TOOL_GOOGLE_DRIVE_ID, TOOL_READ_DOCUMENT_ID, TOOL_SEARCH_FILE_ID } from '@/constants';
```
Comment (abimacarmes) [inline src/interfaces/assistants_web/src/components/Agents/AgentRightPanel.tsx:10]: side note: could be nice to take a look at the constants and organize into data sources vs tools

**Review by abimacarmes — APPROVED** [review]

File: src/interfaces/assistants_web/src/components/Agents/AgentSettings/AgentSettingsForm.tsx (hunk 1 above)
Comment (knajjars) [reply src/interfaces/assistants_web/src/components/Agents/AgentSettings/AgentSettingsForm.tsx:151]: i think showing it at the beginning it is too annoying, might remove it altogether!

File: src/interfaces/assistants_web/src/components/Agents/AgentRightPanel.tsx (hunk 2 above)
Comment (knajjars) [reply src/interfaces/assistants_web/src/components/Agents/AgentRightPanel.tsx:24]: good catch!

File: src/interfaces/assistants_web/src/components/Agents/AgentRightPanel.tsx (hunk 3 above)
Comment (knajjars) [reply src/interfaces/assistants_web/src/components/Agents/AgentRightPanel.tsx:10]: yes agree it is growing quite a bit

**Pull request merged**